rpt.js

```diff
@@ -43,23 +44,24 @@ Node.prototype.path = ''
 Node.prototype.realpath = ''
 Node.prototype.children = null
 
-function Link (pkg, path, realpath, cache) {
-  if (cache[path]) return cache[path]
+function Link (pkg, logical, physical, realpath, cache) {
```

Comment:
So imagine we're installing into `/home/foo/module` and 
`/home` is a symlink to `/mounts/disk3/home`
Further, inside `/home/foo/module` we have:
  `node_modules/bar/…`
  `node_modules/baz -> /home/baz`
When running `read-package-tree` on `/home/foo/module`, we would end up with a tree like:
```json
{
  name: 'module',
  isLink: false,
  path: '/home/foo/module',
  realpath: '/mounts/disk3/home/foo/module',
  children: [
    {
      name: 'bar',
      isLink: false,
      path: '/home/foo/module/node_modules/bar',
      realpath: '/mounts/disk3/home/foo/module/node_modules/bar',
    },
    {
      name: 'baz',
      isLink: true,
      path: '/home/foo/module/node_modules/baz', // is the logical argument
      realpath: '/mounts/disk3/home/baz', // is the realpath argument
      // physical: /mounts/disk3/home/foo/module/node_modules/baz
    }
  ]
}
```